src/models/game_session.ts

```diff
@@ -185,10 +185,20 @@ export default class GameSession {
             return;
         }
 
+        if (guildPreference.getShuffleType() === ShuffleType.UNIQUE && guildPreference.getLimit() === this.lastPlayedSongsQueue.length) {
+            logger.info(`${getDebugContext(message)} | Game session ended (all ${guildPreference.getLimit()} unique songs played)`);
+            await sendEndGameMessage({ channel: message.channel, authorId: message.author.id }, this);
+            await this.endSession();
```

Comment:
Should the game end once all songs have been played? Or restart the queue

Reply:
Either one works, which do you prefer?

Reply:
We could play a different sound or send a message when all songs have played and the queue is resetting